src/fsharp/TypedTreeOps.fs

```diff
@@ -8049,6 +8080,9 @@ let rec typeEnc g (gtpsType, gtpsMethod) ty =
         typarEnc g (gtpsType, gtpsMethod) typar
 
     | TType_measure _ -> "?"
+    | TType_erased_union (_, tys) ->
+        // SWOORUP TODO idk
+        typeEnc g (gtpsType, gtpsMethod) (List.head tys) + "|"
```

Comment:
This is used for XML doc signature key elements in the file generated by `-doc:foo.xml`.  You should use `typeEnc` on the erased representation type

Reply:
(Note that `stripTyEqnsAndMeasureEqns ` has already been called which is erasing some things - we will want to look at extending these erasure routines to cover erasing these unions)